modules/checklist/assets/js/components/progress.js

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react';
+import { Box, LinearProgress, Typography } from '@elementor/ui';
+import PropTypes from 'prop-types';
+
+const Progress = ( { steps } ) => {
+	const [ value, setValue ] = useState( 0 );
+
+	useEffect( () => {
+		const totalSteps = steps.length,
+			doneSteps = steps.filter( ( i ) => i.done ).length;
+
+		setValue( doneSteps / totalSteps * 100 );
```

Comment:
```suggestion
	const progress = getProgressPercentage();

	const getProgressPercentage = () => {
		return steps.map( ( { is_completed } ) => is_completed ) * 100 / steps.length;
	};	
```

Reply:
1. value isn't very suggestive in my opinion
2. I don't see any reason for using useState and useEffect